src/xhity.c

```diff
@@ -3293,9 +3293,16 @@ int flat_acc;
 			/* Houchou always hits when thrown */
 			if (weapon->oartifact == ART_HOUCHOU && thrown)
 				wepn_acc += 1000;
+
 			/* -4 accuracy per weapon size too large (not for thrown objects) */
-			if (!thrown && weapon->objsize - pa->msize > 0){
-				wepn_acc += -4 * (weapon->objsize - pa->msize);
```

Comment:
A 4x also seems to have gone missing here.  The behavior is changed, too, small objects used to get a bonus.  However, that seems wrong and the comment agrees.